### ISSUE TITLE
Mocking  - AttributeError: 'int' object has no attribute 'isdigit'

### ISSUE THREAD

**Issue by jijojv** [cannatag/ldap3#354]
Hi,

I don't see this error on real searches, only when using Mocking using int.

```
In [28]: fake_connection.search(LDAP_GROUP_BASE, '(cn=ops)', attributes=['*'])
Out[28]: True

In [29]: fake_connection.entries
Out[29]:
[DN: cn=ops,cn=groups,dc=example,dc=com - STATUS: Read - READ TIME: 2017-05-28T17:17:42.458575
     cn: ops
     entryDN: cn=ops,cn=groups,dc=example,dc=com
     gidNumber: 123456
     memberUid: jijo1
                jijo2]



In [30]: fake_connection.search(LDAP_GROUP_BASE, '(gidNumber=123456)', attributes=['*'])
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-30-26de748ebf49> in <module>()
----> 1 fake_connection.search(LDAP_GROUP_BASE, '(gidNumber=123456)', attributes=['*'])

/usr/local/lib/python2.7/site-packages/ldap3/core/connection.pyc in search(self, search_base, search_filter, search_scope, dereference_aliases, attributes, size_limit, time_limit, types_only, get_operational_attributes, controls, paged_size, paged_criticality, paged_cookie, auto_escape)
    764             if log_enabled(PROTOCOL):
    765                 log(PROTOCOL, 'SEARCH request <%s> sent via <%s>', search_request_to_dict(request), self)
--> 766             response = self.post_send_search(self.send('searchRequest', request, controls))
    767             self._entries = []
    768

/usr/local/lib/python2.7/site-packages/ldap3/strategy/mockSync.pyc in post_send_search(self, payload)
     64         self.connection.result = dict()
     65         if message_type == 'searchRequest':
---> 66             responses, result = self.mock_search(request, controls)
     67             result['type'] = 'searchResDone'
     68             for entry in responses:

/usr/local/lib/python2.7/site-packages/ldap3/strategy/mockBase.pyc in mock_search(self, request_message, controls)
    567             message = 'incorrect base object'
    568         else:
--> 569             matched = self.evaluate_filter_node(filter_root, candidates)
    570             for match in matched:
    571                 responses.append({

/usr/local/lib/python2.7/site-packages/ldap3/strategy/mockBase.pyc in evaluate_filter_node(self, node, candidates)
    639         if node.elements:
    640             for element in node.elements:
--> 641                 self.evaluate_filter_node(element, candidates)
    642
    643         if node.tag == ROOT:

/usr/local/lib/python2.7/site-packages/ldap3/strategy/mockBase.pyc in evaluate_filter_node(self, node, candidates)
    737             for candidate in candidates:
    738                 # if attr_name in self.connection.server.dit[candidate] and attr_value in self.connection.server.dit[candidate][attr_name]:
--> 739                 if attr_name in self.connection.server.dit[candidate] and self.equal(candidate, attr_name, attr_value):
    740                     node.matched.add(candidate)
    741                 # elif attr_name in self.connection.server.dit[candidate]:  # tries to apply formatters

/usr/local/lib/python2.7/site-packages/ldap3/strategy/mockBase.pyc in equal(self, dn, attribute, value)
    765             formatted_values = [formatted_values]
    766         for attribute_value in formatted_values:
--> 767             if self._check_equality(value, attribute_value):
    768                 return True
    769

/usr/local/lib/python2.7/site-packages/ldap3/strategy/mockBase.pyc in _check_equality(value1, value2)
    772     @staticmethod
    773     def _check_equality(value1, value2):
--> 774         if value1.isdigit() and value2.isdigit():
    775             if int(value1) == int(value2):  # int comparison
    776                 return True

AttributeError: 'int' object has no attribute 'isdigit'
```

**Comment by cannatag** [comment]
Thanks! Your fix will be included in the next release of ldap3.

Bye,
Giovanni

**Comment by cannatag** [comment]
Fixed in release 2.3 of ldap3.
